adagucserverEC/CDFObjectStore.cpp

```diff
@@ -225,6 +225,11 @@ CDFObject *CDFObjectStore::getCDFObject(CDataSource *dataSource, const char *fil
 
 CDFObject *CDFObjectStore::getCDFObject(CDataSource *dataSource, CServerParams *srvParams, const char *fileName, bool plain) {
   CT::string uniqueIDForFile = fileName;
+  // The cdfObject should be unique between files and layers,
+  // otherwise the CDPPOperator datapostproc will not work correctly when using the same file and variable.
+  if (dataSource != NULL && dataSource->layerName.empty() == false) {
+    uniqueIDForFile.concat(dataSource->layerName);
```

Comment:
I am confused what this achieves.

First, the name of the variable is now incorrect.

And it looks like the variable is only used to by added to the files list:
`fileNames.push_back(new CT::string(uniqueIDForFile.c_str()));`

It is not set on the cdfObject (which the comment seems to imply)?

I am sure I am missing something...

Reply:
Explained what this does.

Reply:
In the end I did not like this solution. Reverted it and made an option to enable/disable cdfobject cache via the datareader. The datapostprocessor now creates a reader where the cache is disabled. This way it can safely manipulate the cdfobject without causing issues to other readers reading from the same data object.